modules/MiniFigDirectionDetector.cpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * @file   MinifigDirectionDetector.cpp
+ * @brief  ミニフィグの向きを検出するクラス
+ * @author nishijima515
+ */
+
+#include "MiniFigDirectionDetector.h"
+
+using namespace cv;
+using namespace dnn;
+
+MiniFigDirectionDetector::MiniFigDirectionDetector(const std::string& modelPath)
+{
+  // コンストラクタの実装
+  // ネットワークの読み込み
+  net = readNetFromONNX(modelPath);  // モデルのパスを設定する
+  if(net.empty()) {
+    std::cerr << "モデルの読み込みに失敗しました！" << std::endl;
+  } else {
+    std::cout << "モデルが正常に読み込まれました。" << std::endl;
+  }
+  net.setPreferableBackend(DNN_BACKEND_OPENCV);  // OpenCVバックエンドを使用
+  net.setPreferableTarget(DNN_TARGET_CPU);       // CPUをターゲットに設定
+}
+
+MiniFigDirectionResult MiniFigDirectionDetector::detect(const cv::Mat& frame,
+                                                        const std::string& saveImagePath)
+{
+  MiniFigDirectionResult result;
+
+  // 入力画像の前処理
+  Mat blob = preprocess(frame);
+  net.setInput(blob);
+
+  // ネットワークの推論を実行
+  std::vector<Mat> outputs;
+  net.forward(outputs, net.getUnconnectedOutLayersNames());
+
+  // 出力結果の後処理
+  postprocess(outputs, frame, result, saveImagePath);
+
+  return result;
+}
+
+// 入力画像の前処理を行う関数
+cv::Mat MiniFigDirectionDetector::preprocess(const cv::Mat& frame)
+{
+  cv::Mat blob;
+  blobFromImage(frame, blob, 1.0 / 255.0, Size(640, 640), Scalar(), true, false);
+  return blob;
+}
+
+// 出力結果を後処理して検出結果を生成する関数
+void MiniFigDirectionDetector::postprocess(const std::vector<cv::Mat>& outputs,
+                                           const cv::Mat& frame, MiniFigDirectionResult& result,
+                                           const std::string& saveImagePath)
+{
```

Comment:
このクラスで保存するのは、検出結果のバウンディングボックスとラベルを画像に付与した画像だよね？それなら引数で受け取らなくて、固定のパスでいいと思う。
```suggestion
// 出力結果を後処理して検出結果を生成する関数
void MiniFigDirectionDetector::postprocess(const std::vector<cv::Mat>& outputs,
                                           const cv::Mat& frame, MiniFigDirectionResult& result)
{
```

Reply:
ローカルでテストするときの相対パスと実機で動かすときの相対パスが違うかなと思って、引数でパスを受け取る形にしました。